controllers/etcd_controller.go

```diff
@@ -791,146 +804,167 @@ func (r *EtcdReconciler) getStatefulSetFromEtcd(etcd *druidv1alpha1.Etcd, values
+				{
+					APIVersion:         "druid.gardener.cloud/v1alpha1",
+					BlockOwnerDeletion: pointer.BoolPtr(true),
+					Controller:         pointer.BoolPtr(true),
+					Kind:               "Etcd",
+					Name:               etcd.Name,
+					UID:                etcd.UID,
+				},
 			}
-			return foundEtcd, nil
+			return nil
 		})
-
-		selector, err := metav1.LabelSelectorAsSelector(etcd.Spec.Selector)
-		if err != nil {
-			logger.Error(err, "Error converting etcd selector to selector")
-			return nil, err
-		}
-		dm := NewEtcdDruidRefManager(r.Client, r.Scheme, etcd, selector, etcdGVK, canAdoptFunc)
-
-		logger.Info("Claiming existing cronjob")
-		claimedCronJob, err := dm.ClaimCronJob(ctx, &cronJob)
 		if err != nil {
+			logger.Error(err, "Full snap lease found but error occured during claiming it")
 			return nil, err
 		}
-
-		if _, err = r.syncCronJobSpec(ctx, claimedCronJob, etcd, values, logger); err != nil {
-			return nil, err
-		}
-
-		return claimedCronJob, err
-	}
-
-	// Required cronjob doesn't exist. Create new
-	cj, err := r.getCronJobFromEtcd(etcd, values, logger)
-	if err != nil {
-		return nil, err
-	}
-
-	logger.Info("Creating cronjob", "cronjob", kutil.Key(cj.Namespace, cj.Name).String())
-	err = r.Create(ctx, cj)
-
-	// Ignore the precondition violated error, this machine is already updated
-	// with the desired label.
-	if err == errorsutil.ErrPreconditionViolated {
-		logger.Info("Cronjob precondition doesn't hold, skip updating it", "cronjob", kutil.Key(cj.Namespace, cj.Name).String())
-		err = nil
 	}
-	if err != nil {
-		return nil, err
-	}
-
-	//TODO: Evaluate necessity of claiming object here after creation
-
-	return cj, err
+	return fullLease, nil
 }
 
-func (r *EtcdReconciler) syncCronJobSpec(ctx context.Context, cj *batchv1.CronJob, etcd *druidv1alpha1.Etcd, values map[string]interface{}, logger logr.Logger) (*batchv1.CronJob, error) {
-	decoded, err := r.getCronJobFromEtcd(etcd, values, logger)
-	if err != nil {
-		return nil, err
-	}
-
-	if reflect.DeepEqual(cj.Spec, decoded.Spec) {
-		return cj, nil
-	}
-
-	cjCopy := cj.DeepCopy()
-	cjCopy.Spec = decoded.Spec
+func getFullLease(etcd *druidv1alpha1.Etcd) string {
+	return fmt.Sprintf("%s-full-snap", string(etcd.UID[:6]))
+}
 
-	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		return r.Patch(ctx, cjCopy, client.MergeFrom(cj))
-	})
+func (r *EtcdReconciler) reconcileDeltaLease(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (*coordinationv1.Lease, error) {
+	// Get or Create delta_snapshot_revisions lease object that will keep track of delta snapshot revisions based on which
+	// compaction job will be scheduled
+	nsName := types.NamespacedName{
+		Name:      getDeltaLease(etcd),
+		Namespace: etcd.Namespace,
+	}
+
+	deltaLease := &coordinationv1.Lease{}
+	err1 := r.Get(ctx, nsName, deltaLease)
+	if err1 != nil {
+		logger.Info("Couldn't fetch delta snap lease delta-snapshot-revisions because: " + err1.Error())
```

Comment:
Replace `delta-snapshot-revisions` with actual delta snap lease name here and other occurrences as well